src/Microsoft.Identity.Client/Cache/Keys/MsalAccessTokenCacheKey.cs

```diff
@@ -90,19 +90,20 @@ public override string ToString()
         /// <remarks>
         /// accountId - two guids plus separator - 73 chars        
         /// "accesstoken" string - 11 chars
-        /// env - ussually loging.microsoft.net - 20 chars
+        /// env - a sha256 string - 44 chars
         /// clientid - a guid - 36 chars
         /// tenantid - a guid - 36 chars
         /// scopes - a sha256 string - 44 chars
         /// delimiters - 4 chars
-        /// total: 224 chars
+        /// total: 248 chars
         /// </remarks>
         public string GetUWPFixedSizeKey()
         {
             var crypto = PlatformProxyFactory.GetPlatformProxy().CryptographyManager;
             return MsalCacheCommon.GetCredentialKey(
               _homeAccountId,
-              _environment,
+              crypto.CreateSha256Hash(_environment),
```

Comment:
The actual fix. Because this only affects ATs, we can make changes to the key with minor user impact (i.e. no popups). When adding new tokens, the existing ones with old keys will be deleted by the algorithm that deletes tokens with intersecting scopes.
When reading tokens, we always read all of them, not just the keys.,